code/modules/mob/living/carbon/human/human_powers.dm

```diff
@@ -983,3 +983,39 @@
 
 		visible_message("<span class='notice'>\The [src] attaches \the [O] to \his body!</span>",
 				"<span class='notice'>You attach \the [O] to your body!</span>")
+
+/mob/living/carbon/human/proc/self_diagnostics()
+	set name = "Self-Diagnostics"
+	set desc = "Run an internal self-diagnostic to check for damage."
+	set category = "IC"
+
+	if(stat == DEAD) return
+
+	src << "<span class='notice'>Performing self-diagnostic, please wait...</span>"
+	sleep(50)
```

Comment:
Like Arrow mentioned, use of sleep is terrible because it creates race conditions and for example after sleep is done - IPC might be dead and `stat == DEAD` that is checked before now could be true, but it is not checked after sleep. Either use `do_after()` proc which will make IPC stand still to check their thing and show progress bar. Otherwise you could also use `addtimer()` proc, but then you need to have second proc and put all the code below sleep into this new proc.

Reply:
I don't fully understand what you're asking for here. The verb physically doesn't work and stops running if the IPC is dead. This is fine behavior.
I will fix the <<'s